### PR TITLE
7 date

### PR DESCRIPTION
**Pull request by VitaliiKobrin-in6k**
Save creation and last modify date for devices in db, sort devices by last modify date.

### CODE REVIEW

File: api/src/main/java/com/in6k/vault13/dao/impl/DevicesDaoImpl.java [hunk 1]
```diff
@@ -84,14 +79,17 @@ public Device addEmptyDevice() {
...
     public Device update(Device device) {
-        String query = queryBuilder.updateDevice(device);
-
         try {
-            DaoUtils.getResult(databaseConnector.getArangoDriver(), query, clazz);
+            DocumentEntity<Device> deviceEntity = databaseConnector.getArangoDriver()
+                    .getDocument(ArangoCollections.DEVICES.getName(), device.getId(), Device.class);
+            assignRelations(device);
+            Device deviceForUpdate = new Device();
+            String[] ignoredProperties = {"type", "vendor", "model"};
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/dao/impl/DevicesDaoImpl.java]: Lets use annotation to ignore entity fields.

File: api/src/main/java/com/in6k/vault13/service/impl/DeviceServiceImpl.java [hunk 2]
```diff
@@ -47,13 +48,22 @@ public Device add(Device device) throws ArangoException {
...
+        setDate(device);
         createTypeIfNotExist(device);
         createVendorIfNotExist(device);
         createModelIfNotExist(device);
 
         return devicesDao.update(device);
     }
 
+    private void setDate(Device device) {
+        String currentDate = LocalDateTime.now().toString();
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/service/impl/DeviceServiceImpl.java]: What format does `toString` method use? Is it suitable for time encoding?

File: api/src/main/java/com/in6k/vault13/entity/Device.java [hunk 3]
```diff
@@ -7,6 +7,8 @@
     private Model model;
     private String serialNumber;
     private String notes;
+    private String creationDate;
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/entity/Device.java]: Rename to `createdAt`

File: api/src/main/java/com/in6k/vault13/entity/Device.java [hunk 4]
```diff
@@ -7,6 +7,8 @@
...
+    private String lastModifyDate;
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/entity/Device.java]: Rename to `updatedAt`

File: client/src/app/devices/devices-list.component.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import {Component, OnInit} from '@angular/core';
+import {DateFormatter} from "@angular/common/src/facade/intl";
```
Comment (alexanderko) [inline client/src/app/devices/devices-list.component.ts]: Is there shorter pass to load module?

File: client/src/app/devices/devices-list.component.ts [hunk 6]
```diff
@@ -25,7 +26,7 @@ export class DevicesList implements OnInit {
 
     getDevices(){
         this.deviceService.getDevices()
-            .then(devices => this.devices = devices.sort((a, b) => a.type.name.localeCompare(b.type.name)))
+            .then(devices => this.devices = devices.sort((a, b) => b.lastModifyDate.localeCompare(a.lastModifyDate)))
```
Comment (alexanderko) [inline client/src/app/devices/devices-list.component.ts]: Use [lodash](https://lodash.com) for sorting.

File: api/src/main/java/com/in6k/vault13/service/impl/DeviceServiceImpl.java (hunk 2 above)
Comment (VitaliiKobrin-in6k) [inline api/src/main/java/com/in6k/vault13/service/impl/DeviceServiceImpl.java]: @Override
    public String toString() {
        return date.toString() + 'T' + time.toString();
    }

File: client/src/app/devices/devices-list.component.html [hunk 7]
```diff
@@ -10,6 +10,7 @@
                     <span>{{device.type.name}}</span>
                     <span *ngIf="device.vendor">{{device.vendor.name}}</span>
                     <span *ngIf="device.model">{{device.model.name}}</span>
+                    <span class="date-info">Змiнено: {{getDeviceLastModifyDate(device)}}</span>
```
Comment (alexanderko) [inline client/src/app/devices/devices-list.component.html]: Use [date pipe](https://angular.io/docs/ts/latest/api/common/index/DatePipe-class.html) for formatting.

File: api/src/main/java/com/in6k/vault13/service/impl/DeviceServiceImpl.java (hunk 2 above)
Comment (VitaliiKobrin-in6k) [inline api/src/main/java/com/in6k/vault13/service/impl/DeviceServiceImpl.java]: Result: "2016-08-01T02:20Z"

**Comment by alexanderko** [comment]
@VitaliiKobrin-in6k please fix my comments and do appropriate changes.

File: api/src/main/java/com/in6k/vault13/service/impl/DeviceServiceImpl.java [hunk 8]
```diff
@@ -35,6 +35,7 @@
     @Override
     public List<Device> getAll() throws Exception {
         return devicesDao.getAll().stream()
+                .filter(d -> d.getCreatedAt() != null)
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/service/impl/DeviceServiceImpl.java]: remove this check

File: client/src/app/devices/add-device.component.ts [hunk 9]
```diff
@@ -68,14 +69,14 @@ export class AddDevice implements OnInit {
...
 
         this.modelService
             .getModels()
-            .then(models => this.models = models.sort((a, b) => a.name.localeCompare(b.name)));
+            .then(models => this.models = _.sortBy(models, [m => m.name.toLowerCase()]));
 
         this.typeService
             .getTypes()
-            .then(types => this.types = types.sort((a, b) => a.name.localeCompare(b.name)));
+            .then(types => this.types = _.sortBy(types, [t => t.name.toLowerCase()]));
```
Comment (alexanderko) [inline client/src/app/devices/add-device.component.ts]: extract sorting by name

**Pull request merged**